libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewSubscriber.java

```diff
@@ -31,6 +31,43 @@
   void subscribe(LifecycleOwner owner, final LocationViewModel locationViewModel,
                  final RouteViewModel routeViewModel, final NavigationViewModel navigationViewModel) {
 
+    navigationViewModel.addCallback(new NavigationViewModel.Callback() {
+      @Override
+      public void userOffRoute(OffRouteEvent offRouteEvent) {
+        isOffRoute = true;
```

Comment:
We should keep `public` methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.). What about extracting this block of code into a well-named method? IMO It'll be easier to read and understand and the intention will be clearer.